builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -415,10 +443,12 @@ func resourceArmVirtualMachineCreate(d *schema.ResourceData, meta interface{}) e
 		storageProfile.DataDisks = &dataDisks
 	}
 
+	diagnosticsProfile := expandAzureRmVirtualMachineDiagnosticsProfile(d)
 	networkProfile := expandAzureRmVirtualMachineNetworkProfile(d)
 	vmSize := d.Get("vm_size").(string)
 	properties := compute.VirtualMachineProperties{
-		NetworkProfile: &networkProfile,
+		DiagnosticsProfile: &diagnosticsProfile,
```

Comment:
Will there always be a diagnosticsProfile or should we check that this isn't an empty profile before adding?